### DISCUSSION TITLE
How to enable window.ethereum as provider if available?

### DISCUSSION THREAD

**Discussion by johnpaulcas** [wevm/wagmi#2499] [Q&A]
How to enable the window.ethereum when avialble or its automatic? upon checking there is a code that check if window.ethereum is available or not, is this automatically triggered cause right now we still having an issue 429 when it fallback to default provider or this is fixed on the 1.x version?

```
function getReady(ethereum2) {
          const isMetaMask = !!ethereum2?.isMetaMask;
          if (!isMetaMask)
            return;
          if (ethereum2.isBraveWallet && !ethereum2._events && !ethereum2._state)
            return;
          if (ethereum2.isApexWallet)
            return;
          if (ethereum2.isAvalanche)
            return;
          if (ethereum2.isKuCoinWallet)
            return;
          if (ethereum2.isPortal)
            return;
          if (ethereum2.isTokenPocket)
            return;
          if (ethereum2.isTokenary)
            return;
          return ethereum2;
        }
        if (typeof window === "undefined")
          return;
        const ethereum = window.ethereum;
        if (ethereum?.providers)
          return ethereum.providers.find(getReady);
        return getReady(ethereum);
      },
      ```

**Comment by tmm** [comment]
Not clear on what you are asking. Are you wondering if you can make RPC requests with `window.ethereum`? If so, wagmi doesn't provide an out-of-the-box solution for this since wallets are [not required to support a wide range of RPC methods](https://viem.sh/docs/faq.html#why-doesn-t-wallet-client-support-public-actions).

**Comment by johnpaulcas** [comment]
Hey @tmm , apologies if my previous question was a bit confusing. I'm wondering if we can use window.ethereum as the default way to make transactions on the Ethereum Virtual Machine (EVM) if it's available. Right now, we're using Alchemy, but we keep hitting a rate limit (we get a 429 error). It'd be nice if we could automatically use window.ethereum when it's there, like when a user is connected to Metamask. Does that make sense?
